src/main/java/io/cdap/plugin/dlp/SensitiveRecordRedaction.java

```diff
@@ -288,19 +300,40 @@ private DlpServiceSettings getSettings() throws IOException {
     @Macro
     @Nullable
     private String templateId;
-    private static final Gson gson = new GsonBuilder()
+
+    private static final Gson GSON = new GsonBuilder()
       .registerTypeAdapter(DlpFieldTransformationConfig.class, new DlpFieldTransformationConfigCodec())
       .create();
 
     public List<DlpFieldTransformationConfig> parseTransformations() throws Exception {
-      String[] values = gson.fromJson(fieldsToTransform, String[].class);
+      String[] values = GSON.fromJson(fieldsToTransform, String[].class);
       List<DlpFieldTransformationConfig> transformationConfigs = new ArrayList<>();
       for (String value : values) {
-        transformationConfigs.add(gson.fromJson(value, DlpFieldTransformationConfig.class));
+        transformationConfigs.add(GSON.fromJson(value, DlpFieldTransformationConfig.class));
       }
       return transformationConfigs;
     }
 
+    /**
+     * Get the set of fields that are being transformed or are required for transforms to work. This is used to limit
+     * the payload size to DLP endpoints, the transform will only send the values of the required fields.
+     *
+     * @return Set of field names
+     */
+    public Set<String> getRequiredFields() {
+
+
+      try {
+        HashSet<String> results = new HashSet<>();
+        parseTransformations().stream()
+          .map(DlpFieldTransformationConfig::getRequiredFields)
+          .forEach(results::addAll);
```

Comment:
Use Collector instead of forEach.

```
return parseTransformations().stream()
  .map(DlpFieldTransformationConfig::getRequiredFields)
  .collect(Collectors.toSet());
```

Reply:
Using the collector returns a set of sets which is not what is expected. Is there a better way to flatten a set of sets into one level?

Reply:
Use `flatMap`.

```
return parseTransformations().stream()
  .map(DlpFieldTransformationConfig::getRequiredFields)
  .flatMap(Collection::stream)
  .collect(Collectors.toSet());
```

Reply:
done